specreduce/tracing.py

```diff
@@ -346,6 +346,15 @@ def __post_init__(self):
                 # TODO: implement smoothing with provided width
                 y_bins[i] = ilum2[z_i.argmax()]
 
+        # warn about fully-masked bins (which, currently, means any bin with a single masked value)
+        if len(warn_bins) > 0:
+            plural = 's' if len(warn_bins) > 1 else ''
+            # if there are a ton of bins, we don't want to print them all out
+            if len(warn_bins) > 20:
+                warn_bins = warn_bins[0: 10] + ['...'] + [warn_bins[-1]]
+            warnings.warn(f"All pixels in bin{plural} {', '.join([str(x) for x in warn_bins])}"
```

Comment:
Could do the `plural` logic inline (and remove it from earlier)... might be more readable, but up to you.

```suggestion
            warnings.warn(f"All pixels in {'bins' if len(warn_bins) else 'bin'} {', '.join([str(x) for x in warn_bins])}"
```